src/cache/cache_operations.rs

```diff
@@ -20,12 +25,68 @@ pub struct Invalidate {
     database: Database,
 }
 
+#[derive(Deserialize, Debug)]
+pub enum InvalidationScope {
+    Requests,
+    Responses,
+    Both,
```

Comment:
Wondering if we can get away without `Both` but have something like:

```javascript
{
    "invalidate" :"Transactions",
    "pattern_details":  ["0xd6f5Bef6bb4acD235CF85c0ce196316d10785d67","Responses,Requests"]
}
```

to invalidate `Both`. Maybe this increases the complexity of the implementation that we have too 🤔

Reply:
Right, in that case we would have to somehow tokenise the string and check every entry in the array against the enum, which as it is, we get for free if we use individual values.